### PR TITLE
[Feature #159236198] Create users via REST API

### PR DESCRIPTION
**Pull request by wasswa-derick**
#### What does this PR do?
* Allow users/apps to create users via REST API

#### Description of Task to be completed?
*  Create a permission for REST API user creation
* Admin grants permissions to the users to create users via the REST API. 
* Add special flags to the user profile to identify users created via the REST API or not.
* Create model to show REST API users relationship with their creators.
* Create a serializer for the user objects
* Create Viewset for REST API user creation
* Create the private API endpoint for user registration
* List all REST API users

#### How should this be manually tested?
* Clone the project, ```https://github.com/andela/wg-wits.git```.
* In your terminal, ```cd wg-wits```
* Checkout to the ```ft-api-create-users-159236198``` branch.
* In the project directory, create a virtual environment and activate it: run ```virtualenv -p python3 venv-django``` then ```source venv-django/bin/activate```
* Install requirements ```pip install -r requirements_devel.txt```
* Run ```python manage.py migrate``` to  include the new migrations.
* Serve the application ```python manage.py runserver```
* Run the command ```python manage.py add-user-rest-api 'username or email' --username_or_email``` to grant permissions to a user to create users.
* In the browser, visit the endpoint for the users with, ```http://127.0.0.1:8000/api/v2/users/```
* This shall list all the users available.
* In the input fields at the bottom of the screen, enter data in the fields and post.
* To list the users create via REST API, run this command ```python manage.py list-user-rest-api```

#### What are the relevant pivotal tracker stories?
[#159236198](https://www.pivotaltracker.com/story/show/159236198)

#### Screenshots (if appropriate)

* Accessing user creation without permission

screen shot 2018-08-16 at 11 07 15

* Granting user creation permission to a user

screen shot 2018-08-16 at 11 08 43

* Users Endpoint

screen shot 2018-08-16 at 11 05 59

* Creating a user

screen shot 2018-08-16 at 11 06 35

* User Created

screen shot 2018-08-16 at 11 06 48

* List REST API users

screen shot 2018-08-16 at 11 09 13

### CODE REVIEW

File: wger/core/management/commands/add-user-rest-api.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    def handle(self, *position_args, **options):
+        # Emphasize that the command should provide atleast one of these options
+        if options.get('username') is False and options.get('email') is False:
+            self.stdout.write('Option for the username or email must be specified !!!')
+            return
+
+        # Check if the user submitted email
+        if options.get('email'):
+            email = str(position_args[0])
+            user = User.objects.get(email=email)
```
Comment (RrNn) [inline wger/core/management/commands/add-user-rest-api.py]: Catch User Doesn't Exist exception

**Comment by solnsubuga** [comment]
@wasswa-derick Please ensure apps cannot create users with already existing emails.

File: wger/core/management/commands/add-user-rest-api.py (hunk 1 above)
Comment (RrNn) [reply wger/core/management/commands/add-user-rest-api.py]: Replace the 3 exclamation marks with one and don't put a space between the last letter and the exclamation mark.

File: wger/core/management/commands/list-user-rest-api.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        api_users = []
+        for user in wger_users:
+            if user.userprofile.rest_api_user:
+                api_users.append(user)
+
+        if len(api_users) == 0:
+            self.stdout.write('No user has not been created via REST API.')
+
+        for user in api_users:
+            self.stdout.write('Username: {} , Email: {}' . format(user.username, user.email))
```
Comment (RrNn) [inline wger/core/management/commands/list-user-rest-api.py]: `Email: {}' . format` Get rid of that space

**Review by RrNn — CHANGES_REQUESTED** [review]

**Comment by solnsubuga** [comment]
@wasswa-derick please update the how to manually test this section to include the new instructions.

**Review by solnsubuga — APPROVED** [review]
Looks good.

**Comment by Yiga-fred** [comment]
LGTM

**Review by RrNn — APPROVED** [review]
LGTM

**Pull request merged**